### ISSUE TITLE
line number missing when switching between buffers

### ISSUE THREAD

**Issue by karta0807913** [redguardtoo/emacs.d#1070]
I've noticed a significant issue: when rapidly switching between buffers, the line numbers disappear. This can be quite disruptive to the workflow.

I suspect this happens because the code snippet below is not correctly bound to the buffer:

https://github.com/redguardtoo/emacs.d/blob/0673ea80be6a17df2883bc0ca5d634b368d80d21/lisp/init-linum-mode.el#L48-L54

I have a quick fix ready. Could we create a pull request to update this Lisp file? 

```lisp
(defun my-setup-line-number-mode ()
  "Set up line number display."
  ;; use idler to speed up emacs startup
  (unless (or (memq major-mode my-linum-inhibit-modes)
              ;; don't show line number for certain file extensions
              (my-should-use-minimum-resource))
    (let ((buffer (current-buffer)))
      (my-run-with-idle-timer 0.5 (lambda ()
                                    (with-current-buffer buffer
                                      (display-line-numbers-mode)))))))
```

Thank you!

**Comment by redguardtoo** [comment]
Thanks.  I just removed idle timer which is legacy code for old version of Emacs.

9ae587f3 fixed #1070 line number missing when switching between buffers (Chen Bin)

**Comment by karta0807913** [comment]
Awesome
